### DISCUSSION TITLE
Automatic bank imports

### DISCUSSION THREAD

**Discussion by Lucaber** [TheAxelander/OpenBudgeteer#251] [Ideas]
Hi,
im using OpenBugeteer for over two years now and have build multiple scripts to automatically import transaction from my bank accounts.

Im pulling new transactions from my bank accounts every hour. Im using the `libfintx` library for accounts which support the hbci/fints protocol (mostly german banks). For other accounts im calling a custom external bash script, with mostly curl calls.
The transactions are then imported into the database with a slightly more complicated duplicate matcher, because im also importing pending transaction of which the transaction date and memo can change. The database also has an extra field to store whether the transaction is pending.
For every new transaction im also sending a notification (currently via an email) to myself.

When I started implementing there was no api available, so i used a separate c# service project inside the openbudgeteer solution and imported the types and database connection.
My current setup is very hacky (eg. with hardcoded passwords) but im interested in publishing a more polished future version.
Now, im asking whether you are interested in integrating these features directly into OpenBugeteer or if I should create a separate project?
These features could be useful for a lot of other users, especially if they are configurable via the web interface.

**Comment by TheAxelander** [comment]
Happy to see this and very interested on some integrations. Something like that would need to be of course tested and maintained also in future which was one of the reasons why I was hesitated to touch this automatic bank import feature. Especially after my decision on stopping the support on SQLite, where I just had gave up on it, I want to include things now, that I can maybe also support in future, even if certain contributors are leaving at a certain point in time.

As there is also now an API available (I still need to get some experience on that too, to do it "the right way"), I think it's quite easy for you to start in a separate project with that. Kind of like a plugin, separate Docker container, script, whatever...

Depending on the code complexity level, we can still decide later to fully integrate this or not. But for the beginning I think it's more convenient for you to have your own space and release cycle (as I tend to have a longer release cycle spending more time on testing new features on my day-by-day usage).

Of course I can do any kind of reference in Readme, Documentation, etc. to your project. 

As said, I'm very happy to see that someone puts also effort into this project and I'm open for any kind of contribution/interaction.
